engines/order_management/app/services/order_management/order/updater.rb

```diff
@@ -234,7 +234,10 @@ def requires_authorization?
       end
 
       def update_pending_payment
-        return unless order.state.in? ["payment", "confirmation"]
+        # We only want to update complete order pending payment when it's a cash payment. We assume
+        # that if the payment was a credit card it would alread have been processed, so we don't
+        # bother checking the payment type
```

Comment:
Assumptions worry me, what if they result in an even more obscure bug in the future? It's based on another part of the system that could perhaps change. 
But I'm not sure exactly what change to suggest so happy to leave this as it is ;)